### PR TITLE
Open3D-ML Jupyter Tutorials

### PR DESCRIPTION
**Pull request by AlexZakIntel**
Sanskar, Sameer, and Benjamin,

This PR is for newly created **Open3D-ML Jupyter Tutorials**. They were run from the `Open3D-ML/docs/tutorial/notebook/` path.

Please feel free to critique or suggest changes or improvements.

Thank you all very much!

---
This change is [Reviewable](https://reviewable.io/reviews/isl-org/open3d-ml/512)

### CODE REVIEW

File: docs/tutorial/notebook/Inference_on_a_custom_data.ipynb [hunk 1]
```diff
@@ -0,0 +1,333 @@
+{
```
Comment (ssheorey) [inline docs/tutorial/notebook/Inference_on_a_custom_data.ipynb:1]: <p>Running Semantic Segmentation inference on custom data</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/isl-org/Open3D-ML/pull/512/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (ssheorey) [inline docs/tutorial/notebook/Inference_on_a_custom_data.ipynb:1]: <ul><li>This needs an introduction at the top describing what we will do in this tutorial.</li><li>The language describing the steps is not quite right. @sanskar, you may need to directly edit Alex's PR. e.g: (-in our data model, we define a dataset....)</li><li>The note at the top about downloading weights is specific to PyTorch. Add TF alternative.</li></ul><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/isl-org/Open3D-ML/pull/512/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (ssheorey) [inline docs/tutorial/notebook/Inference_on_a_custom_data.ipynb:1]: <p>... to the <s>data</s> weights...</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/isl-org/Open3D-ML/pull/512/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (ssheorey) [inline docs/tutorial/notebook/Inference_on_a_custom_data.ipynb:1]: <p>Add (commented) command to download the weights directly in the notebook:</p><p> </p><p><code> # from urllib.request import urlretrieve</code></p><p><code> # urlretrieve(weights_url, filename=weights_file)</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/isl-org/Open3D-ML/pull/512/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (ssheorey) [inline docs/tutorial/notebook/Inference_on_a_custom_data.ipynb:1]: <p>Explain <code>in_channels=3</code></p><p>...with our <s>data</s> weights file...</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/isl-org/Open3D-ML/pull/512/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (ssheorey) [inline docs/tutorial/notebook/Inference_on_a_custom_data.ipynb:1]: <p>...display what <s>the</s> <s>content</s> <code>data...</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/isl-org/Open3D-ML/pull/512/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

File: docs/tutorial/notebook/reading_a_config_file.ipynb [hunk 2]
```diff
@@ -0,0 +1,608 @@
+{
```
Comment (ssheorey) [inline docs/tutorial/notebook/reading_a_config_file.ipynb:1]: <p>Change the goal of this tutorial to:</p><p><strong>Controlling training and inference with config files</strong></p><p> </p><p>The tutorial should answer these questions for a user:</p><ul><li>How do I specify where my data is?</li><li>How do I check the current training or inference parameters? How do I change them?</li><li>How do command line arguments and config file parameters interact? When should either be used and which has precedence?</li></ul><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/isl-org/Open3D-ML/pull/512/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (ssheorey) [inline docs/tutorial/notebook/reading_a_config_file.ipynb:1]: <p>open3d.ml.torch - Open3D-ML Torch API library</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/isl-org/Open3D-ML/pull/512/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (ssheorey) [inline docs/tutorial/notebook/reading_a_config_file.ipynb:1]: <p>Checking the type is not useful. Instead use</p><p><code>vars(cfg) </code></p><p>This will list the 3 dictionaries in cfg</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/isl-org/Open3D-ML/pull/512/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (ssheorey) [inline docs/tutorial/notebook/reading_a_config_file.ipynb:1]: <p>Mention that items can be viewed as well as updated like in a standard Python dict. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/isl-org/Open3D-ML/pull/512/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

File: docs/tutorial/notebook/reading_a_dataset.ipynb [hunk 3]
```diff
@@ -0,0 +1,373 @@
+{
```
Comment (ssheorey) [inline docs/tutorial/notebook/reading_a_dataset.ipynb:1]: <p>Only keep the part of the image inside the red box - that's clearer than the entire image + arrow.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/isl-org/Open3D-ML/pull/512/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (ssheorey) [inline docs/tutorial/notebook/reading_a_dataset.ipynb:1]: <p>Make this image smaller - the font size should match the font size in the text.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/isl-org/Open3D-ML/pull/512/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

File: docs/tutorial/notebook/train_ss_model_using_pytorch.ipynb [hunk 4]
```diff
@@ -0,0 +1,333 @@
+{
```
Comment (ssheorey) [inline docs/tutorial/notebook/train_ss_model_using_pytorch.ipynb:1]: <p>This file needs an editing pass for language and organization. Also, rephrase from a user's perspective who is trying to learn what functionality is available and exactly what it will do. As an example, mention the inputs needed to make the main steps work (e.g. how do you get the dataset?) and what is the result. e.g. after running training, you should have a checkpoint (where?) that you can use for testing or inference.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/isl-org/Open3D-ML/pull/512/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (ssheorey) [inline docs/tutorial/notebook/train_ss_model_using_pytorch.ipynb:1]: <p>In this tutorial, we will learn how to train a semantic segmentation model using PyTorch<s> in a Jupyter Notebook. We assume that you are familiar with Jupyter Notebook and have created a folder notebooks in a folder that is relative to ml3d. </s></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/isl-org/Open3D-ML/pull/512/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (ssheorey) [inline docs/tutorial/notebook/train_ss_model_using_pytorch.ipynb:1]: <p>Needs editing to remove repeating language:</p><p> </p><blockquote>Run inference with a pre-trained model. For this example, we will use the <code>RandLANet</code> model.</blockquote><p>Rephrase:</p><blockquote>Train a model. We will train a model using the <code>SemanticKITTI</code> dataset and <code>RandLANet</code> model.</blockquote><p>Rephrase for clarity:</p><blockquote>Run an inference and run a test. We will run an inference using the 'training' split that use a pointcloud and display a result. However, a test is run on a pre-defined test set rather than a pass pointcloud.</blockquote><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/isl-org/Open3D-ML/pull/512/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (ssheorey) [inline docs/tutorial/notebook/train_ss_model_using_pytorch.ipynb:1]: <blockquote> For this example, we will use the <code>RandLANet</code> model.</blockquote><p>This is already covered earlier. Avoid repetition.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/isl-org/Open3D-ML/pull/512/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (ssheorey) [inline docs/tutorial/notebook/train_ss_model_using_pytorch.ipynb:1]: <blockquote><s>Running a test is very similar to training the model. </s></blockquote><p>Instead, say what this will do:</p><p>Next we will evaluate the trained model on the test split.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/isl-org/Open3D-ML/pull/512/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

File: docs/tutorial/notebook/Inference_on_a_custom_data.ipynb (hunk 1 above)
Comment (AlexZakIntel) [reply docs/tutorial/notebook/Inference_on_a_custom_data.ipynb:1]: <p>Done.</p>

Comment (AlexZakIntel) [reply docs/tutorial/notebook/Inference_on_a_custom_data.ipynb:1]: <p>Done.</p>

Comment (AlexZakIntel) [reply docs/tutorial/notebook/Inference_on_a_custom_data.ipynb:1]: <p>Done.</p>

File: docs/tutorial/notebook/reading_a_config_file.ipynb (hunk 2 above)
Comment (AlexZakIntel) [reply docs/tutorial/notebook/reading_a_config_file.ipynb:1]: <p>Done.</p>

File: docs/tutorial/notebook/reading_a_dataset.ipynb (hunk 3 above)
Comment (AlexZakIntel) [reply docs/tutorial/notebook/reading_a_dataset.ipynb:1]: <p>Done.</p>

Comment (AlexZakIntel) [reply docs/tutorial/notebook/reading_a_dataset.ipynb:1]: <p>Done.</p>

File: docs/tutorial/notebook/train_ss_model_using_pytorch.ipynb (hunk 4 above)
Comment (AlexZakIntel) [reply docs/tutorial/notebook/train_ss_model_using_pytorch.ipynb:1]: <p>Done.</p>

Comment (AlexZakIntel) [reply docs/tutorial/notebook/train_ss_model_using_pytorch.ipynb:1]: <p>Done.</p>

File: docs/tutorial/notebook/Inference_on_a_custom_data.ipynb (hunk 1 above)
Comment (AlexZakIntel) [reply docs/tutorial/notebook/Inference_on_a_custom_data.ipynb:1]: <p>Done.</p>

Comment (AlexZakIntel) [reply docs/tutorial/notebook/Inference_on_a_custom_data.ipynb:1]: <p>Done</p>

File: docs/tutorial/notebook/reading_a_config_file.ipynb (hunk 2 above)
Comment (AlexZakIntel) [reply docs/tutorial/notebook/reading_a_config_file.ipynb:1]: <p>Done.</p>

File: docs/tutorial/notebook/train_ss_model_using_pytorch.ipynb (hunk 4 above)
Comment (AlexZakIntel) [reply docs/tutorial/notebook/train_ss_model_using_pytorch.ipynb:1]: <p>Done.</p>

Comment (AlexZakIntel) [reply docs/tutorial/notebook/train_ss_model_using_pytorch.ipynb:1]: <p>Done.</p>

Comment (AlexZakIntel) [reply docs/tutorial/notebook/train_ss_model_using_pytorch.ipynb:1]: <p><strong>To Do (Sanskar):</strong> editing pass for language and organization</p>

File: docs/tutorial/notebook/reading_a_config_file.ipynb (hunk 2 above)
Comment (AlexZakIntel) [reply docs/tutorial/notebook/reading_a_config_file.ipynb:1]: <p><strong>To Do </strong>- Sanskar</p>

File: docs/tutorial/notebook/Inference_on_a_custom_data.ipynb (hunk 1 above)
Comment (AlexZakIntel) [reply docs/tutorial/notebook/Inference_on_a_custom_data.ipynb:1]: <p><strong>To Do (Sanskar):</strong> editing pass</p>

Comment (sanskar107) [reply docs/tutorial/notebook/Inference_on_a_custom_data.ipynb:1]: <p>Done</p>

File: docs/tutorial/notebook/train_ss_model_using_pytorch.ipynb (hunk 4 above)
Comment (sanskar107) [reply docs/tutorial/notebook/train_ss_model_using_pytorch.ipynb:1]: Done

File: docs/tutorial/notebook/reading_a_config_file.ipynb (hunk 2 above)
Comment (sanskar107) [reply docs/tutorial/notebook/reading_a_config_file.ipynb:1]: Adding information about command line arguments in this file doesn't make sense as this tutorial is intended for reading config file through python code, and we don't talk about our launch script anywhere.

I am planning to add a new tutorial on using our launch script `scripts/run_pipeline.py`, which will include the usage of command line arguments, precedences, distributed training etc..
